src/main/java/com/redhat/devtools/lsp4ij/client/features/EditorBehaviorFeature.java

```diff
@@ -176,9 +179,59 @@ public static boolean enableTextMateNestedBracesImprovements(@NotNull PsiFile fi
      * @return true if the semantic tokens-based file view provider is enabled; otherwise false
      */
     public static boolean enableSemanticTokensFileViewProvider(@NotNull PsiFile file) {
-        return LanguageServiceAccessor.getInstance(file.getProject()).hasAny(
-                file.getVirtualFile(),
-                ls -> ls.getClientFeatures().getEditorBehaviorFeature().isEnableSemanticTokensFileViewProvider(file)
+        return isEnabled(file, EditorBehaviorFeature::isEnableSemanticTokensFileViewProvider);
+    }
+
+    private interface FeatureFlagChecker {
+        boolean isEnabled(@NotNull EditorBehaviorFeature editorBehaviorFeature, @NotNull PsiFile file);
+    }
+
+    private static boolean isEnabled(
+            @NotNull PsiFile file,
+            @NotNull FeatureFlagChecker featureFlagChecker
+    ) {
+        return isEnabled(file.getProject(), file.getVirtualFile(), featureFlagChecker);
+    }
+
+    private static boolean isEnabled(
+            @NotNull Project project,
+            @Nullable VirtualFile virtualFile,
+            @NotNull FeatureFlagChecker featureFlagChecker
+    ) {
+        if (virtualFile == null) return false;
+
+        Key<CachedValue<Boolean>> cacheKey = getCacheKey(virtualFile, featureFlagChecker);
+        return CachedValuesManager.getManager(project).getCachedValue(
+                project,
+                cacheKey,
+                () -> {
+                    PsiFile file = LSPIJUtils.getPsiFile(virtualFile, project);
+                    if ((file == null) || !file.isValid()) {
+                        // If we couldn't find the PSI file, return false/disabled with no caching
+                        return Result.create(false, ModificationTracker.EVER_CHANGED);
```

Comment:
As it says, if we couldn't resolve a PSI file -- which shouldn't really ever happen, but I'm being defensive -- return `false`/disabled, but with no caching so that we can try again the next time we're called.